akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -166,6 +165,20 @@ object Sink {
   }
 
   /**
+    * A `Sink` that counts incoming elements until upstream terminates.
+    * As upstream may be unbounded, `Flow[T].take` or the stricter `Flow[T].limit` (and their variants)
+    * may be used to ensure boundedness.
+    * Materializes into `CompletionStage` of `[[IOResult]]` containing value of the counter and information
+    * about the outcome of streaming.
+    * Standard behavior of `Long` overflowing applies.
+    *
+    * See also [[Flow.limit]], [[Flow.limitWeighted]], [[Flow.take]], [[Flow.takeWithin]], [[Flow.takeWhile]]
+    */
+  def count[In]: Sink[In, CompletionStage[IOResult]] = {
```

Comment:
I see that `IOResult` is exactly the type we need, but the name `IO` feels wrong. Should we introduce a duplicate `CountResult`? Otherwise the doc descriptions in `IOResult` should be generalized.